cpp/include/cugraph/algorithms.hpp

```diff
@@ -1350,6 +1350,123 @@ random_walks(raft::handle_t const& handle,
              bool use_padding                                     = false,
              std::unique_ptr<sampling_params_t> sampling_strategy = nullptr);
 
+/**
+ * @brief returns uniform random walks from starting sources, where each path is of given
+ * maximum length.
+ *
+ * Note: starting sources can contain duplicates, in which case different random walks will
+ *       be generated for each instance.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
+ * @tparam weight_t Type of edge weights. Needs to be a floating point type.
+ * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view graph view to operate on
+ * @param start_span Device span defining the starting vertices
+ * @param max_depth maximum length of random walk
+ * @param seed (optional, defaults to system time), seed for random number generation
+ * @return tuple containing device vectors of vertices and the edge weights<br>
+ *         For each input selector there will be (max_depth+1) elements in the
+ *         vertex vector with the starting vertex followed by the subsequent
+ *         vertices in the random walk.  If a path terminates before max_depth,
+ *         the vertices will be populated with invalid_vertex_id
+ *         (-1 for signed vertex_t, std::numeric_limits<vertex_t>::max() for an
+ *         unsigned vertex_t * type)<br>
+ *         For each input selector there will be max_depth elements in the weights
+ *         vector with the edge weight for the edge in the path.  If a path
+ *         terminates before max_depth the subsequent edge weights will be
+ *         set to weight_t{0}.
+ */
+// TODO: Do I care about transposed or not?  I want to be able to operate in either
+// direction.
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
+std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<weight_t>> uniform_random_walks(
+  raft::handle_t const& handle,
+  graph_view_t<vertex_t, edge_t, weight_t, false, multi_gpu> const& graph_view,
+  raft::device_span<vertex_t const> start_span,
+  size_t max_depth,
+  uint64_t seed = 0);
```

Comment:
How does this function work for unweighted graphs? Still return rmm::device_uvector<weight_t>?

Reply:
I don't know that we've explored support for an unweighted graph very much.

I could wrap the return in `std::optional` and return `std::nullopt` if the graph is unweighted.  `biased_*` would fail on an unweighted graph.  `node2vec_*` could assume a weight of 1 on an unweighted graph.

Reply:
Gotcha, and I think it is better to specify how we handle unweighted graphs in the documentation.

Reply:
Updated in next push (check all 3 APIs)